source/features/action-pr-link.tsx

```diff
@@ -2,25 +2,49 @@ import * as pageDetect from 'github-url-detection';
 
 import features from '../feature-manager.js';
 import observe from '../helpers/selector-observer.js';
+import {getConversationNumber} from '../github-helpers/index.js';
 
-async function add(prLink: HTMLAnchorElement): Promise<void> {
+function setSearchParameter(anchorElement: HTMLAnchorElement, name: string, value: string): void {
+	const parameters = new URLSearchParams(anchorElement.search);
+	parameters.set(name, value);
+	anchorElement.search = String(parameters);
+}
+
+async function addForRepositoryActions(prLink: HTMLAnchorElement): Promise<void> {
 	const prNumber = prLink.textContent!.slice(1);
 
 	const runLink = prLink.closest('.Box-row')!.querySelector('a.Link--primary')!;
-	const parameters = new URLSearchParams(runLink.search);
-	parameters.set('pr', prNumber);
-	runLink.search = String(parameters);
+	setSearchParameter(runLink, 'pr', prNumber);
+}
+
+async function addForPR(actionLink: HTMLAnchorElement): Promise<void> {
+	const prNumber = getConversationNumber()?.toString();
+
+	if (prNumber === undefined) {
+		return;
+	}
+
+	setSearchParameter(actionLink, 'pr', prNumber);
 }
 
-async function init(signal: AbortSignal): Promise<void> {
-	observe('div.Box-row[id^=check_suite_] a[data-hovercard-type="pull_request"]', add, {signal});
+async function initForRepositoryActionsPage(signal: AbortSignal): Promise<void> {
+	observe('div.Box-row[id^=check_suite_] a[data-hovercard-type="pull_request"]', addForRepositoryActions, {signal});
+}
+
+async function initForPRPage(signal: AbortSignal): Promise<void> {
+	observe(':not(span.rgh-ci-link) > details.commit-build-statuses div.merge-status-item:has(a[href="/apps/github-actions"]) a.status-actions', addForPR, {signal});
```

Comment:
Still too specific. The tag name is rarely needed. We usually specify it only when we want to get a specific tag in typescript, for example in this case we specify the selected `a` element.

This works

```suggestion
	// Exclude rgh-link, include isPRCommits
	observe('main [href="/apps/github-actions"] ~ div a.status-actions', addForPR, {signal});
```

Reply:
Looks much better :D